python-avd/pyavd/_eos_designs/shared_utils/misc.py

```diff
@@ -364,3 +364,107 @@ def l3_bgp_neighbors(self: SharedUtilsProtocol) -> list:
+            if intf_with_rss:
+                append_if_not_duplicate(
+                    list_of_dicts=rss_profile_member_interfaces,
+                    primary_key="name",
+                    new_dict=intf_with_rss,
+                    context="Ethernet interface defined with RSS settings",
+                    context_keys=["name"],
+                    ignore_same_dict=False,
+                )
+        # iterate thru member interfaces of each L3 Port-Channel
+        for interface in self.node_config.l3_port_channels:
+            for member_intf in interface.member_interfaces:
+                intf_with_rss = self.build_interface_with_rx_queue_settings(member_intf)
+                if intf_with_rss:
+                    append_if_not_duplicate(
+                        list_of_dicts=rss_profile_member_interfaces,
+                        primary_key="name",
+                        new_dict=intf_with_rss,
+                        context="Port-Channel member interface defined with RSS settings",
+                        context_keys=["name"],
+                        ignore_same_dict=False,
+                    )
+        if rss_profile_member_interfaces:
+            return rss_profile_member_interfaces
+        return None
+
+    def build_interface_with_rx_queue_settings(
+        self: SharedUtilsProtocol,
+        l3_member_interface: EosDesigns._DynamicKeys.DynamicNodeTypesItem.NodeTypes.NodesItem.L3InterfacesItem
+        | EosDesigns._DynamicKeys.DynamicNodeTypesItem.NodeTypes.NodesItem.L3PortChannelsItem.MemberInterfacesItem,
+    ) -> dict | None:
+        """Returns interface dictionary with RSS settings."""
+        intf_with_rss_settings = {}
+        if l3_member_interface._get("rx_queue"):
+            # validate rx_queue 'count' when specified
+            if l3_member_interface.rx_queue._get("count") and l3_member_interface.rx_queue.count > self.max_rss_queues:
+                msg = f"'rx_queue' count for interface '{l3_member_interface.name}' exceeds maximum supported '{self.max_rss_queues}' for this platform."
+                raise AristaAvdInvalidInputsError(msg)
+            intf_with_rss_settings["name"] = l3_member_interface.name
+            rx_queue_worker_cum_set = set()
+            for _worker in l3_member_interface.rx_queue.worker:
+                for worker_id in range_expand(_worker):
+                    if int(worker_id) >= self.max_rss_queues:
+                        msg = (
+                            f"One or more worker ids within '{_worker}' under 'rx_queue' for interface '{l3_member_interface.name}' "
+                            f"equal or exceed maximum supported '{self.max_rss_queues}' for this platform."
+                        )
+                        raise AristaAvdInvalidInputsError(msg)
+                    rx_queue_worker_cum_set.add(int(worker_id))
+            rx_queue_worker_cum = list_compress(sorted(rx_queue_worker_cum_set))
```

Comment:
it is getting sorted inside list_compress, but we need it to be a list.
```suggestion
            rx_queue_workers_range = list_compress(list(rx_queue_workers))
```